### PR TITLE
[USM] Removed agent-usm.jar from system-probe binary to reduce deployment binary size

### PR DESCRIPTION
**Pull request by val06**
### What does this PR do?

Since Java TLS is not supported yet, removing the java agent from system-probe container (reduces ~26MB) 

### Motivation

Reduce binary size of our deployment

### Possible Drawbacks / Trade-offs

After this change, if someone enables the JavaTLS flag, the init phase of the javaTLS ebpf program will fail, which will cause general failure of `initProtocols` (since we operate all or nothing atm). 
Talked about it with @guyarb 

### Describe how to test/QA your changes

Shouldn't affect the existing UTs as they use a fake java agent to test the functionality

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: tasks/process_agent.py [hunk 1]
```diff
@@ -127,7 +127,7 @@ def build_dev_image(ctx, image=None, push=False, base_image="datadog/agent:lates
         ctx.run(f"chmod 0444 {docker_context}/*.o {docker_context}/*.c {docker_context}/co-re/*.o")
         ctx.run(f"cp /opt/datadog-agent/embedded/bin/clang-bpf {docker_context}")
         ctx.run(f"cp /opt/datadog-agent/embedded/bin/llc-bpf {docker_context}")
-        ctx.run(f"cp pkg/network/protocols/tls/java/agent-usm.jar {docker_context}")
+
```
Comment (guyarb) [inline tasks/process_agent.py:130]: I think we should undo the change - this task (and the Dockerfile) you changed are meant to patch a dev-image via a manual command line. So the change has no impact on the actual artifacts we provide for customers, and it is still required for the testing purposes

File: tasks/system_probe.py [hunk 2]
```diff
@@ -794,9 +794,6 @@ def kitchen_prepare(ctx, windows=is_windows, kernel_release=None, ci=False, pack
             if os.path.isdir(extra_path):
                 shutil.copytree(extra_path, os.path.join(target_path, extra))
 
-        if pkg.endswith("java"):
-            shutil.copy(os.path.join(pkg, "agent-usm.jar"), os.path.join(target_path, "agent-usm.jar"))
-
```
Comment (guyarb) [inline tasks/system_probe.py:799]: AFAIK, the change in the file will make the CI to fail
It will not impact our docker size

**Review by guyarb — APPROVED** [review]
LGTM

**Review by chouquette — APPROVED** [review]
